packages/node_modules/@ciscospark/plugin-phone/test/integration/spec/events-model.js

```diff
@@ -0,0 +1,659 @@
+          await sleep(1000);
+          assert.equal(sc.me.state, `connected`, `spock expects to stay connected after mccoy leaves`);
+          assert.equal(sc.state, `active`, `spock expects the call to remain active after mccoy leaves`);
+
+          await Promise.all([
+            expectMembershipDisconnectedEvent(sc, spock.id, `spock expects spock to disconnect`),
+            expectInactiveEvent(sc, `spock expects the call to go inactive after leaving the call`),
+            expectInactiveEvent(mc, `mccoy expects the call to go inactive after spock leaves the call`),
+            sc.hangup()
+          ]);
+
+          assert.equal(sc.me.state, `disconnected`, `spock expects to be disconnected after spock leaves`);
+          assert.equal(findMembership(mc, spock).state, `disconnected`, `mccoy expects to be disconnected after spock leaves`);
+          assert.equal(sc.state, `inactive`, `spock expects the call to go inactive after spock leaves`);
+          assert.equal(mc.state, `inactive`, `mccoy expects the call to go inactive after spock leaves`);
+        }));
+      });
+
+      describe(`a declined two party room call`, () => {
+        let mccoy, room, spock;
+        const users = {
+          mccoy: null,
+          spock: null
+        };
+
+        setup(users, (r) => {
+          mccoy = users.mccoy;
+          spock = users.spock;
+          room = r;
+        });
+
+        it(`proceeds through a series of events`, () => handleErrorEvent(spock.spark.phone.dial(room.id), async (sc) => {
+          // This is more to prove our assumptions about the test than to require
+          // a particular code behavior.
+          assert.notOk(sc.locus, `We never expect the call to have a locus on the first tick; if this assertion does not hold, it is a sign the test suite assumptions were incorrect`);
+
+          const [mc] = await Promise.all([
+            expectCallIncomingEvent(mccoy.spark.phone, `mccoy expects to receive a call notification`),
+            expectChangeLocusEvent(sc, `spock expects to receive a locus`)
+              .then(() => {
+                assert.equal(sc.state, `active`, `spock's call goes active when the locus arrives`);
+                assert.equal(sc.me.state, `connected`, `spock's participant entry goes active when the locus arrives`);
+                assert.equal(findMembership(sc, mccoy).state, `notified`, `spock expects mccoy to be notified`);
+              })
+          ]);
+
+          assert.equal(mc.me.state, `notified`);
+          assert.equal(findMembership(mc, spock).state, `connected`, `mccoy expects spock to be be connected when mccoy receives the call`);
+
+          await Promise.all([
+            expectMembershipDeclinedEvent(sc, mccoy.id, `spock expects mccoy to decline the call`),
+            expectMembershipDeclinedEvent(mc, mccoy.id, `mccoy expects mccoy to decline the call`),
+            // expectDeclinedEvent(mc, `mccoy expects to decline`),
```

Comment:
Commented line needed?

Reply:
Yea, I'll add some comments on these. These are things we should be asserting, but can't at this time because we shutdown the event listeners when you disconnect. Before or as part of 2.0, I want to change the call event model so that calls can be rejoined (and, therefore, we won't turn the listeners off until it's appropriate). Unfortunately, that depends on understanding our memory leak issue first.